singer_sdk/testing/tap_tests.py

```diff
@@ -126,7 +126,7 @@ def test(self) -> None:
         )
         stream_record_keys = set().union(*(d.keys() for d in self.stream_records))
         diff = stream_transformed_keys - stream_record_keys
-        if diff:
+        if stream_record_keys and diff:
```

Comment:
Do you think returning early if `self.stream_records` is empty would also work here?